### PR TITLE
Adds Auth Guard

### PR DESCRIPTION
**Pull request by shudsonbstar4**
DO NOT MERGE

I want to get thoughts before we incorporate this (and not step on Cristina's toes).

### CODE REVIEW

File: src/app/hikes/hikes-routing.module.ts [hunk 1]
```diff
@@ -21,7 +25,7 @@ const routes: Routes = [
       },
       {
         path: ':id',
-        component: HikeDetailComponent,
+        component: HikeDetailComponent
```
Comment (vjwilson) [inline src/app/hikes/hikes-routing.module.ts]: Arrrrrgh! 
Must... have... comma... dangle... or... won't... be... able... to... breathe (gasp)

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
 
-  constructor() { }
+  constructor(
+    private router: Router,
+  ) { }
 
   ngOnInit() {
   }
 
+  login() {
```
Comment (vjwilson) [inline src/app/home/home.component.ts:17]: Nice low-tech way to simulate login

**Review by bellizio — APPROVED** [review]

File: src/app/hikes/hikes-routing.module.ts (hunk 1 above)
Comment (shudsonbstar4) [reply src/app/hikes/hikes-routing.module.ts]: not sure if for real or... LOL

Do you want me to add them there? It's one of those things that should probably be added to an editorconfig b/c I'm not consistent with that

Comment (vjwilson) [reply src/app/hikes/hikes-routing.module.ts]: This is the first I noticed we were doing it differently. I've been using the AirBnB JS styleguide lint rules for so long, I didn't even check the Angular CLI default lint config.

I guess that is tslint.json?

Looks like it's loosy-goosey by default, with no trailing-comma rule setting.

**Comment by shudsonbstar4** [comment]
@vjwilson I think this is ready. If you agree, let's 🚢 it

**Pull request merged**